cluster/failover_config.go

```diff
@@ -92,12 +95,19 @@ func (c *FailoverConfig) Validate(rootConfig Config) error {
 	if len(c.Configs) == 0 {
 		return fmt.Errorf("at least one failover cluster config expected: %w", hzerrors.ErrIllegalArgument)
 	}
-	for i, cc := range c.Configs {
-		if err := cc.Validate(); err != nil {
+	for i := range c.Configs {
+		ccp := &c.Configs[i]
+		// if a cluster connection timeout was not specified, set it to default.
+		t := ccp.ConnectionStrategy.Timeout
+		if t == 0 {
+			t = types.Duration(internal.DefaultConnectionTimeoutWithFailover)
+		}
+		if err := ccp.Validate(); err != nil {
 			return err
 		}
+		ccp.ConnectionStrategy.Timeout = t
```

Comment:
This is updating users config right ? 
I think we should avoid changing what user provided. It should be read only anywhere internal to the client. 
This way, it will be easier to reason. Changing the users config caused really unexpected bugs in the past for java client.

Reply:
It wasn't supposed to, but it did, due to validation of the main config instead of a clone of it. Fixed in: https://github.com/hazelcast/hazelcast-go-client/pull/615/commits/bd77cd8b5727b5cbfd93ebe26473f2ea38becb19